dali/kernels/test/warp_test/warp_gpu_test.cu

```diff
@@ -0,0 +1,393 @@
+  if (HasFailure()) {
+    cv::imwrite("Warp_Affine_RotateScale_out.png", cv_out);
+    cv::imwrite("Warp_Affine_RotateScale_ref.png", cv_ref);
+    cv::Mat diff;
+    cv::absdiff(cv_out, cv_ref, diff);
+    cv::imwrite("Warp_Affine_RotateScale_diff.png", diff);
+  }
+}
+
+
+TEST(WarpGPU, Affine_RotateScale_Uniform) {
+  cv::Mat cv_img = cv::imread(testing::dali_extra_path() + "/db/imgproc/dots.png");
+  auto cpu_img = view_as_tensor<uint8_t>(cv_img);
+  auto gpu_img = copy<AllocType::GPU>(cpu_img);
+  auto img_tensor = gpu_img.first;
+
+  vec2 center(cv_img.cols * 0.5f, cv_img.rows * 0.5f);
+
+  const int samples = 10;
+  std::vector<AffineMapping2D> mapping_cpu(samples);
+  int scale = 10;
+
+  TensorListView<StorageGPU, uint8_t, 3> in_list;
+  in_list.resize(samples, 3);
+  for (int i = 0; i < samples; i++) {
+    in_list.shape.set_tensor_shape(i, img_tensor.shape);
+    in_list.data[i] = img_tensor.data;
+
+    auto tr = translation(center) * rotation2D(-2*M_PI * i / samples) *
+              translation(-center) * scaling(vec2(1.0f/scale, 1.0f/scale));
+    mapping_cpu[i] = sub<2, 3>(tr, 0, 0);
+  }
+
+  WarpGPU<AffineMapping2D, 2, uint8_t, uint8_t, uint8_t, DALI_INTERP_LINEAR> warp;
+
+  ScratchpadAllocator scratch_alloc;
+
+  auto mapping_gpu = memory::alloc_unique<AffineMapping2D>(AllocType::GPU, samples);
+  TensorShape<2> out_shape = { img_tensor.shape[0] * scale, img_tensor.shape[1] * scale };
+  KernelContext ctx = {};
+  std::vector<TensorShape<2>> out_shapes_hw(samples);
+  for (int i = 0; i < samples; i++)
+    out_shapes_hw[i] = out_shape;
+  auto mappings = make_tensor_gpu<1>(mapping_gpu.get(), { samples });
+  copy(mappings, make_tensor_cpu<1>(mapping_cpu.data(), { samples }));
+
+  auto out_shapes = warp.GetOutputShape(in_list.shape, make_span(out_shapes_hw));
+  KernelRequirements req = warp.WarpSetup::Setup(out_shapes, true);
+  scratch_alloc.Reserve(req.scratch_sizes);
+  TestTensorList<uint8_t, 3> out;
+  out.reshape(req.output_shapes[0].to_static<3>());
+  auto scratchpad = scratch_alloc.GetScratchpad();
+  ctx.scratchpad = &scratchpad;
+  warp.Run(ctx, out.gpu(0), in_list, mappings, make_span(out_shapes_hw), 255);
+  cudaDeviceSynchronize();
+
+  for (int i = 0; i < samples; i++) {
+    auto cpu_out = out.cpu(0)[i];
+    ASSERT_EQ(cpu_out.shape[0], out_shapes_hw[i][0]);
+    ASSERT_EQ(cpu_out.shape[1], out_shapes_hw[i][1]);
+    ASSERT_EQ(cpu_out.shape[2], 3);
+
+    cv::Mat cv_out(cpu_out.shape[0], cpu_out.shape[1], CV_8UC3, cpu_out.data);
+
+    cv::Matx<float, 2, 3> cv_transform = AffineToCV(mapping_cpu[i]);
+
+    cv::Mat cv_ref;
+    cv::warpAffine(cv_img, cv_ref,
+                  cv_transform, cv::Size(out_shape[1], out_shape[0]),
+                  cv::INTER_LINEAR|cv::WARP_INVERSE_MAP,
+                  cv::BORDER_CONSTANT, cv::Scalar(255, 255, 255, 255));
+    auto ref_img = view_as_tensor<uint8_t>(cv_ref);
+    Check(cpu_out, ref_img, EqualEps(8));
+    char out_name[64];
```

Comment:
Maybe this dumping could be generalized and reused in Affine_RotateScale_Single as well.